src/wallet/rpcwallet.cpp

```diff
@@ -6241,12 +6391,16 @@ UniValue z_getnotescount(const UniValue& params, bool fHelp)
     if (!EnsureWalletIsAvailable(fHelp))
         return NullUniValue;
 
-    if (fHelp || params.size() > 1)
+    if (fHelp || params.size() > 2)
         throw runtime_error(
-            "z_getnotescount\n"
+            "z_getnotescount ( minconf asOfHeight )\n"
+            "\nReturns the number of notes available in the wallet for each shielded value pool.\n"
             "\nArguments:\n"
             "1. minconf      (numeric, optional, default=1) Only include notes in transactions confirmed at least this many times.\n"
-            "\nReturns the number of notes available in the wallet for each shielded value pool.\n"
+            "2. asOfHeight       (numeric, optional) Execute the query as if it were run when\n"
+            "                    the blockchain was at the height specified by this argument.\n"
+            "                    The default is to use the entire blockchain that the node is\n"
+            "                    aware of.\n"
```

Comment:
```suggestion
            "2. asOfHeight   (numeric, optional) Execute the query as if it were run when\n"
            "                the blockchain was at the height specified by this argument.\n"
            "                The default is to use the entire blockchain that the node is\n"
            "                aware of.\n"
```
Document what happens if `asOfHeight` is greater than the height of the last known block.

Reply:
Reformatting no longer relevant.